File: DarkOrbit 10.0/Net/netty/commands/ClientUIMenuBarModule.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Ow.Utils;

namespace Ow.Net.netty.commands
{
    class ClientUIMenuBarModule
    {
        public static short NOT_ASSIGNED = 0;
        public static short GENERIC_FEATURE_BAR = 2;
        public static short GAME_FEATURE_BAR = 1;

        public static short ID = 21788;

        public String mPosition = "";
        public short mMenuId = 0;
        public List<ClientUIMenuBarItemModule> mMenuBarItems;
        public String var_792 = "";

        public ClientUIMenuBarModule(short param1, List<ClientUIMenuBarItemModule> param2, string param3, string param4)
        {
            this.mMenuId = param1;
            this.mMenuBarItems = param2;
            this.mPosition = param3;
            this.var_792 = param4;
        }

        public byte[] write()
        {
            var param1 = new ByteArray(ID);
            param1.writeShort(this.mMenuId);
            param1.writeUTF(this.var_792);
            param1.writeUTF(this.mPosition);
            param1.writeInt(this.mMenuBarItems.Count);
            foreach(var mbi in this.mMenuBarItems)
            {
                param1.write(mbi.write());
            }
            return param1.Message.ToArray();
        }
    }
}

File: DarkOrbit 9.0/DarkOrbit 9.0 Emulator/src/simulator/netty/serverCommands/ClientUIMenuBarModule.java
package simulator.netty.serverCommands;

import java.io.DataOutputStream;
import java.io.IOException;
import java.util.Vector;

import simulator.netty.ServerCommand;

public class ClientUIMenuBarModule
        implements ServerCommand {

    public static short NOT_ASSIGNED        = 0;
    public static short GENERIC_FEATURE_BAR = 2;
    public static short GAME_FEATURE_BAR    = 1;

    public static int ID = 7181;

    public String mPosition = "";
    public short  mMenuId   = 0;
    public Vector<ClientUIMenuBarItemModule> mMenuBarItems;
    public String var_792 = "";

    public ClientUIMenuBarModule(String pPosition, short pMenuId, Vector<ClientUIMenuBarItemModule> pMenuBarItems,
                                 String param4) {
        this.mPosition = pPosition;
        this.mMenuId = pMenuId;
        this.mMenuBarItems = pMenuBarItems;
        this.var_792 = param4;
    }

    public int getID() {
        return ID;
    }

    public int method_1005() {
        return 8;
    }

    public void write(DataOutputStream out) {
        try {
            out.writeShort(ID);
            this.writeInternal(out);
        } catch (IOException e) {
            e.printStackTrace();
        }
    }

    protected void writeInternal(DataOutputStream out) {
        try {
            out.writeShort(-10485);
            out.writeUTF(this.mPosition);
            out.writeShort(-15203);
            out.writeUTF(this.var_792);
            out.writeInt(this.mMenuBarItems.size());
            for (ClientUIMenuBarItemModule menuBarItem : this.mMenuBarItems) {
                menuBarItem.write(out);
            }
            out.writeShort(this.mMenuId);
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}